canopy_driver.F90

```diff
@@ -92,27 +92,28 @@ program canopy_driver
 ! ... read canopy profile data that should be passed
       open(9,  file='input_profile.txt',  status='old')
       i0 = 0
-      read(9,*,iostat=i0)  	! skip headline
+      read(9,*,iostat=i0)  ! skip headline
       do i=1, canlays
         read(9, *) profile(i)
       end do
 ! ... initialize canopy model and integrate to get fractional plant area distribution functions
       zkcm   = profile%zk 
       ztothc = zkcm/hcm 
       resz   = profile%dzk
-      cansublays  = hcm/resz(canlays) 
+      cansublays  = ceiling(hcm/resz(canlays))
       canmidpoint = cansublays/2
-      flamelays    = flameh/resz(canlays)
+      flamelays    = ceiling(flameh/resz(canlays))
       midflamepoint   = flamelays/2
  ! calculate canopy/foliage distribution shape profile - bottom up total in-canopy and fraction at z
+      fainc = 0  ! initialize
       do i=1, canlays
         if (ztothc(i) >= zcanmax .and. ztothc(i) <= 1.0) then
            fainc(i) = exp((-1.0*((ztothc(i)-zcanmax)**2.0))/sigmau**2.0)
-        else if (ztothc(i) >= 0.0 .and. ztothc(i) <= zcanmax) then
+        else if (ztothc(i) >= 0.0 .and. ztothc(i) < zcanmax) then
```

Comment:
I see that it is that way in the paper. But we *should* make a choice of what we want it to be in the case that `ztothc(i)` is exactly equal to `zcanmax`. This change actually has no effect, since `ztothc(i) >= zcanmax` comes first, it just makes it more clear what our choice is for the case of `ztothc(i)` exactly equal to `zcanmax`.

Reply:
@zmoon can you please change this back?

Reply:
ok